src/node.js

```diff
@@ -215,8 +215,30 @@
   startup.globalVariables = function() {
     global.process = process;
     global.global = global;
-    global.GLOBAL = global;
-    global.root = global;
+
+    // Deprecate GLOBAL and root
+    ['GLOBAL', 'root'].forEach(function(name) {
+      // getter
+      const get = NativeModule.require('util').deprecate(function() {
```

Comment:
Tiny suggestion: the code might look a little better if you pulled the `NativeModule.require('util')` out into a variable. Feel free to ignore if you want.